packages/core-app-api/src/app/AppRouter.tsx

```diff
@@ -71,13 +71,14 @@ function SignInPageWrapper({
   const [identityApi, setIdentityApi] = useState<IdentityApi>();
   const configApi = useApi(configApiRef);
   const basePath = getBasePath(configApi);
+  const optionalSignOut = configApi.getOptionalString('app.signOutUrl');
 
   if (!identityApi) {
     return <Component onSignInSuccess={setIdentityApi} />;
   }
 
   appIdentityProxy.setTarget(identityApi, {
-    signOutTargetUrl: basePath || '/',
+    signOutTargetUrl: optionalSignOut || basePath || '/',
```

Comment:
Huge bonus points if you could add a test for this if it's not too complicated 🙏

Reply:
There is a test here in the [AppIdentityProxy.test.ts](https://github.com/backstage/backstage/blob/master/packages/core-app-api/src/apis/implementations/IdentityApi/AppIdentityProxy.test.ts#L83) that sets the `signOutTargetUrl` to `/foo`. But if need be I can try to add a test under the `AppRouter`.

Reply:
Separate discussion, but the intention of the `signOutTargetUrl` is for it to be the redirect URL after sign-out, not triggering the sign-out itself. Is it possible to send a request in the background to oauth2 proxy instead? Just a fetch to the logout endpoint with credentials. I'm thinking that should trigger the sign-out too. I'm hoping that with that we don't need the redirect on their end either, just the native one we have here.

For that we'd need the client-side request added [here](https://github.com/backstage/backstage/blob/23aece85fc32c093f2c758662f488e311ce3218a/packages/core-components/src/layout/ProxiedSignInPage/ProxiedSignInIdentity.ts#L141), and the logout backend endpoint implemented over [here](https://github.com/backstage/backstage/blob/23aece85fc32c093f2c758662f488e311ce3218a/plugins/auth-backend/src/providers/oauth2-proxy/provider.ts#L89), as a redirect. Might be that we need a bit of extra logic in the client/server communication there to make the request happen from the client so that it hits the oauth2 proxy endpoint.

Reply:
Ah, I understand. I did try that route originally before stumbling upon the `signOutTargetUrl`. My biggest issue was coming from constructing the url `/oauth2/sign_out?rd=http%3A%2F%2Fmy-provider.com%2Fend_session_endpoint` . To bypass the logout page, the id token needs to be passed as an additional parameters. That and the many redirects that was happening through the requests was causing some headaches. But I can definetly attempt another try and report back any issues that I run into.

Reply:
Yep I see the issue with the provider logout. One potential route there is that it must be provided as config to the oauth2 provider in the auth backend, and then we refer to the [OAuth2 Proxy docs](https://oauth2-proxy.github.io/oauth2-proxy/docs/features/endpoints/) for how to find out what to set there.